src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -105,6 +109,36 @@
      */
     private final Cache<String, ActiveDirectoryGroupDetails> groupCache;
 
+    /**
+     *  Ldap extra properties added in Active Directory Configure Global Security UI
+     */
+    private List<ActiveDirectorySecurityRealm.EnvironmentProperty> extraEnvVars;
+
+    /**
+     * Properties to be passed to the current LDAP context
+     */
+    private Hashtable<String, String> props = new Hashtable<String, String>();
+
+    /**
+     * Timeout if no connection after 30 seconds
+     */
+    protected final static String DEFAULT_LDAP_CONNECTION_TIMEOUT = "30000";
```

Comment:
I think these belong in the Descriptor.  Then let the descriptor set the property in bind if it is not set (if it has not been explicitly configured, to the System property value (if present) or the default (if sys prop is not present))